app/src/main/java/com/example/inventory/ui/item/ItemEditScreen.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.inventory.ui.item
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.example.inventory.InventoryTopAppBar
+import com.example.inventory.R
+import com.example.inventory.ui.AppViewModelProvider
+import com.example.inventory.ui.navigation.NavigationDestination
+import com.example.inventory.ui.theme.InventoryTheme
+import kotlinx.coroutines.launch
+
+object ItemEditDestination : NavigationDestination {
+    override val route = "item_edit"
+    override val titleRes = R.string.edit_item_title
+    const val itemIdArg = "itemId"
+    val routeWithArgs = "$route/{$itemIdArg}"
+}
+
+@Composable
+fun ItemEditScreen(
+    navigateBack: () -> Unit,
+    onNavigateUp: () -> Unit,
+    modifier: Modifier = Modifier,
```

Comment:
modifier should be the last

Reply:
No, it should be the first optional parameter.

Reply:
is viewModel optional?